### PR TITLE
Fix: #474 Generated Policy Size Too Long

### PR DESCRIPTION
**Pull request by StewartW**
*Issue #, if available:* #474

*Description of changes:*
[Broken down in the gist here](https://gist.github.com/StewartW/aff84e6f86811b128ca03b4d2014f559)

Separates out the policies for cloudformation-role in the target accounts and codepipeline role in the deployment account. 
Now have policies for S3, KMS and a base policy that catches everything else. 

Updated the shared IAM library to take in a list of policies to update now, also accepts single string and will treat it accordingly. 
Existing update policy code also provides backward compatibility because it checks if the statement with the SID exists in the policy and if it can't find it - it does nothing. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/lambda_codebase/initial_commit/bootstrap_repository/adf-bootstrap/deployment/global.yml [hunk 1]
```diff
@@ -516,7 +525,7 @@ Resources:
               - kms:ReEncrypt*
             Resource: !GetAtt KMSKey.Arn
       Roles:
-        - !Ref CloudFormationRole
+        - !Ref CloudFormationRole
```
Comment (sbkok) [inline src/lambda_codebase/initial_commit/bootstrap_repository/adf-bootstrap/deployment/global.yml:528]:
```suggestion
        - !Ref CloudFormationRole
```

File: src/lambda_codebase/initial_commit/bootstrap_repository/adf-bootstrap/deployment/global.yml [hunk 2]
```diff
@@ -938,6 +916,67 @@ Resources:
...
+              - s3:List*
+              - s3:Put*
+              - s3:ReplicateDelete
+              - s3:ReplicateObject
+              - s3:ReplicateTags
+            Resource:
+              - !Sub arn:${AWS::Partition}:s3:::${PipelineBucket}
+              - !Sub arn:${AWS::Partition}:s3:::${PipelineBucket}/*
+      Roles:
+        - !Ref CodePipelineRole
```
Comment (sbkok) [inline src/lambda_codebase/initial_commit/bootstrap_repository/adf-bootstrap/deployment/global.yml:979]:
```suggestion
        - !Ref CodePipelineRole
```

File: src/lambda_codebase/initial_commit/bootstrap_repository/adf-bootstrap/deployment/global.yml [hunk 3]
```diff
@@ -938,6 +916,67 @@ Resources:
...
+            Action:
+              - kms:Decrypt
+              - kms:DescribeKey
+              - kms:Encrypt
+              - kms:GenerateDataKey*
+              - kms:ReEncrypt*
+            Resource: !GetAtt KMSKey.Arn
+      Roles:
+        - !Ref CodePipelineRole
+  CodePipelineRolePolicySTS:
```
Comment (sbkok) [inline src/lambda_codebase/initial_commit/bootstrap_repository/adf-bootstrap/deployment/global.yml:938]:
```suggestion
  CodePipelineRolePolicySTS:
```

File: src/lambda_codebase/initial_commit/bootstrap_repository/adf-bootstrap/deployment/global.yml [hunk 4]
```diff
@@ -938,6 +916,67 @@ Resources:
             Resource: "*"
       Roles:
         - !Ref CodePipelineRole
+  CodePipelineRolePolicyKMS:
```
Comment (sbkok) [inline src/lambda_codebase/initial_commit/bootstrap_repository/adf-bootstrap/deployment/global.yml:919]:
```suggestion
  CodePipelineRolePolicyKMS:
```

File: src/lambda_codebase/initial_commit/bootstrap_repository/adf-bootstrap/global.yml [hunk 5]
```diff
@@ -117,6 +117,24 @@ Resources:
...
+            Sid: "S3"
+            Action:
+              - s3:Get*
+              - s3:List*
+              - s3:Put*
+            Resource:
+              - !Sub arn:${AWS::Partition}:s3:::${DeploymentAccountBucketName}
+              - !Sub arn:${AWS::Partition}:s3:::${DeploymentAccountBucketName}/*
+      Roles:
+        - !Ref CloudFormationRole
```
Comment (sbkok) [inline src/lambda_codebase/initial_commit/bootstrap_repository/adf-bootstrap/global.yml:137]:
```suggestion
        - !Ref CloudFormationRole
```

**Review by sbkok — CHANGES_REQUESTED** [review]
Only a few trailing space issues to be fixed. Except those it looks good, thanks!

**Review by javydekoning — APPROVED** [review]

**Review by sbkok — APPROVED** [review]
Thanks!

**Pull request merged**